clients/ethereum/client_test.go

```diff
@@ -695,13 +643,118 @@ func TestClient_ExecuteTransfer(t *testing.T) {
 			},
 		}
 
-		hash, err := c.ExecuteTransfer(context.Background(), common.Hash{}, batch, 5)
+		hash, err := c.ExecuteTransfer(context.Background(), common.Hash{}, argLists, batch.ID, 5)
 		assert.Equal(t, "0xc5b2c658f5fa236c598a6e7fbf7f21413dc42e2a41dd982eb772b30707cba2eb", hash)
 		assert.Nil(t, err)
 		assert.True(t, wasCalled)
 	})
 }
 
+func TestClient_CheckRequiredBalance(t *testing.T) {
+	args := createMockEthereumClientArgs()
+
+	tokenErc20 := common.BytesToAddress([]byte("ERC20token1"))
+	balance := big.NewInt(1000000)
+	t.Run("not enough erc20 balance", func(t *testing.T) {
+		c, _ := NewEthereumClient(args)
+		c.erc20ContractsHandler = &bridgeTests.ERC20ContractsHolderStub{
+			BalanceOfCalled: func(ctx context.Context, erc20Address common.Address, address common.Address) (*big.Int, error) {
+				assert.Equal(t, c.safeContractAddress, address)
+
+				return balance, nil
+			},
+		}
+		err := c.CheckRequiredBalance(context.Background(), tokenErc20, big.NewInt(0).Add(balance, big.NewInt(1)))
+		assert.True(t, errors.Is(err, errInsufficientErc20Balance))
+	})
+	t.Run("erc20 balance of errors", func(t *testing.T) {
+		expectedErr := errors.New("expected error erc20 balance of")
+		c, _ := NewEthereumClient(args)
+		c.erc20ContractsHandler = &bridgeTests.ERC20ContractsHolderStub{
+			BalanceOfCalled: func(ctx context.Context, erc20Address common.Address, address common.Address) (*big.Int, error) {
+				return nil, expectedErr
+			},
+		}
+
+		err := c.CheckRequiredBalance(context.Background(), tokenErc20, balance)
+		assert.True(t, errors.Is(err, expectedErr))
+	})
+	t.Run("should work", func(t *testing.T) {
+		c, _ := NewEthereumClient(args)
+		c.erc20ContractsHandler = &bridgeTests.ERC20ContractsHolderStub{
+			BalanceOfCalled: func(ctx context.Context, erc20Address common.Address, address common.Address) (*big.Int, error) {
+				assert.Equal(t, c.safeContractAddress, address)
+
+				return balance, nil
+			},
+		}
+		err := c.CheckRequiredBalance(context.Background(), tokenErc20, balance)
+		assert.Nil(t, err)
+	})
+}
+
+// TokenMintedBalances
+// WhitelistedTokensMintBurn
+func TestClient_TokenMintedBalances(t *testing.T) {
+	t.Run("error while getting token minted balances", func(t *testing.T) {
+		expectedErr := errors.New("expected error")
```

Comment:
extract it globally and reuse the same expectedErr

Reply:
in other context expectedErr has other value